pkg/resource/deploy/source_eval.go

```diff
@@ -2197,7 +2233,25 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 			Options: options,
 		})
 		if err != nil {
-			return nil, err
+			if st, ok := status.FromError(err); ok {
+				// We only turn errors into diagnostics for a few specific status codes that the grpc library doesn't generate.
```

Comment:
What do we mean by "doesn't generate" here? That the SDKs won't handle them, or that something else elsewhere won't, for instance?